config variables not expanded in cache-from

It seems like the config variables are not expanded in cache-from

```yaml
meta:
  project: project

image=server:
  image: {project}
  pull-base-image-on-build: false
  target: server
  cache-from:
    - {project}:latest   ##### <==== seems to have no effect, cache is never used.
  context: .
  tags:
    - latest
```

When I replace `{project}` with `project` (the expansion) manually dobi uses cache for build.

okay, I know nothing about go development but followed some guide, ended up with cloning the project into `github.com/dnephin/dobi` inside my `$GOPATH` and then I run `go get`, it downloaded a bunch of stuff and then failed at

```
~/dev/go/src/github.com/dnephin/dobi$ go get -v
# github.com/dnephin/dobi
/usr/lib/go-1.10/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/tmp/go-link-414185749/000024.o: In function `_cgo_f404b13b0081_Cfunc_sysconf':
/tmp/go-build/cgo-gcc-prolog:32: multiple definition of `_cgo_f404b13b0081_Cfunc_sysconf'
/tmp/go-link-414185749/000018.o:/tmp/go-build/cgo-gcc-prolog:32: first defined here
collect2: error: ld returned 1 exit status
```

```
$ go version
go version go1.10.4 linux/amd64
```

any pointers? I'd like to take a stab at this.  I like the project and there are still some things I'd like to add, so why not I guess :D

--- a/config/image.go
+++ b/config/image.go
@@ -151,6 +151,11 @@ func (c *ImageConfig) Resolve(resolver Resolver) (Resource, error) {
 		return &conf, err
 	}
 
+	conf.CacheFrom, err = resolver.ResolveSlice(c.CacheFrom)
+	if err != nil {
+		return &conf, err
+	}
+
 	conf.Image, err = resolver.Resolve(c.Image)
 	if err != nil {
 		return &conf, err

--- a/config/image.go
+++ b/config/image.go
@@ -15,7 +15,7 @@ import (
 // ImageConfig An **image** resource provides actions for working with a Docker
 // image. If an image is buildable it is considered up-to-date if all files in
 // the build context have a modified time older than the created time of the
-// image. If using inline Dockerfile, the **dobi.yaml** file will be considered 
+// image. If using inline Dockerfile, the **dobi.yaml** file will be considered
 // as a part of the build context.
 // name: image
 // example: An image with build args:

--- a/config/image_test.go
+++ b/config/image_test.go
@@ -113,6 +113,7 @@ func TestImageConfigResolve(t *testing.T) {
 			"key1": "{one}",
 			"key2": "ok",
 		},
+		CacheFrom: []string{"{one}", "two"},
 	}
 	resolved, err := image.Resolve(resolver)
 	assert.NilError(t, err)
@@ -124,6 +125,7 @@ func TestImageConfigResolve(t *testing.T) {
 			"key1": "thetag",
 			"key2": "ok",
 		},
+		CacheFrom: []string{"thetag", "two"},
 	}
 	assert.Check(t, is.DeepEqual(expected, resolved, cmpConfigOpt))
 }

--- a/dockerfiles/Dockerfile.build
+++ b/dockerfiles/Dockerfile.build
@@ -1,7 +1,7 @@
-FROM    golang:1.10-alpine
+FROM    golang:1.13-alpine
 RUN     apk add -U git bash curl tree
 
-ARG     FILEWATCHER_SHA=v0.3.0
+ARG     FILEWATCHER_SHA=v0.3.2
 RUN     go get -d github.com/dnephin/filewatcher && \
         cd /go/src/github.com/dnephin/filewatcher && \
         git checkout -q "$FILEWATCHER_SHA" && \
@@ -15,7 +15,7 @@ RUN     go get -d github.com/golang/dep/cmd/dep && \
         go build -v -o /usr/bin/dep ./cmd/dep && \
         rm -rf /go/src/* /go/pkg/* /go/bin/*
 
-ARG     GOTESTSUM=v0.3.0
+ARG     GOTESTSUM=v0.4.0
 RUN     go get -d gotest.tools/gotestsum && \
         cd /go/src/gotest.tools/gotestsum && \
         git checkout -q "$GOTESTSUM" && \

--- a/dockerfiles/Dockerfile.docs
+++ b/dockerfiles/Dockerfile.docs
@@ -1,4 +1,4 @@
-FROM    alpine:3.6
+FROM    alpine:3.10
 
 RUN     apk -U add \
             python \

--- a/tasks/job/build.go
+++ b/tasks/job/build.go
@@ -328,7 +328,9 @@ func createFromTar(tarReader io.Reader, header *tar.Header, path artifactPath) e
 		return os.Symlink(header.Linkname, hostPath)
 
 	default:
-		logging.Log.Warnf("Unhandled file type from archive %s: %s", header.Typeflag, header.Name)
+		logging.Log.Warnf("Unhandled file type from archive %s: %s",
+			string(header.Typeflag),
+			header.Name)
 	}
 
 	return nil

--- a/tasks/job/capture.go
+++ b/tasks/job/capture.go
@@ -57,6 +57,6 @@ func (t *captureTask) Run(ctx *context.ExecuteContext, depsModified bool) (bool,
 		return false, nil
 	}
 
-	logging.ForTask(t).Debug("Setting %q to: %s", t.variable, out)
+	logging.ForTask(t).Debugf("Setting %q to: %s", t.variable, out)
 	return true, os.Setenv(t.variable, out)
 }

